src/dashboard/components/modal.py

```diff
@@ -1,4 +1,16 @@
-"""Module containing functionality related to modals."""
+"""Module containing functionality related to modals.
+
+When creating a modal, a modal container should be created
+in the page first. A modal dialog can then be passed
+as a child to the modal container. When the open property
+is modified on the child, the modal container will react
+to the change and will show or hide depending on the state.
+
+Example:
```

Comment:
Improper code block formatting. Should be:
```
Example::

    code here
```
https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks